workers/unity/Packages/io.improbable.worker.sdk/IOStorage.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using Improbable.Worker.CInterop.Internal;
+
+namespace Improbable.Worker.CInterop
+{
+    public class IOStorage : IDisposable
+    {
+        private readonly CIO.StorageHandle storage;
+
+        public IOStorage()
+        {
+            storage = CIO.StorageCreate();
+        }
+
+        /// <inheritdoc cref="IDisposable"/>
+        public void Dispose()
+        {
+            storage.Dispose();
+            GC.SuppressFinalize(this);
+        }
```

Comment:
Couple comments on the implementation of `Dispose()` here and in `IOStream`:

- We are calling `GC.SuppressFinalize(this);`, but we don't have a destructor defined (`~IOStorage()`). I think you may be getting a compiler warning about this.. if you mark this class as `sealed` it will fix that warning as well. 
- If a user called `Dispose()` twice, we would get double-free errors. We should track if this object has already been disposed and no-op the second (or subsequent) times after that.

Reply:
> We should track if this object has already been disposed and no-op the second (or subsequent) times after that.

The worker SDK doesn't seem to do this as far as I can see. If they have already called dispose on the object, won't that mean any state in the class could have been freed already?

Reply:
Hmmm maybe its handled inside the `CPtrHandle` class which the `CIO.StreamHandle` inherits from.

Reply:
It doesn't look like it's handled in `CptrHandle`. The only place I see something like this is when the `WrappedGcHandle` is used and the `IsAllocated` flag is checked before the handle is disposed.

Reply:
Maybe they just trust that anyone who inherits from this class won't try and double-free the memory?